### PR TITLE
[RFC] transition to modern dailink

### PR DESCRIPTION
**Pull request by plbossart**
This is WIP to get feedback only on changes in machine drivers. These patches are based on Mark Brown's tree, they'd need to be cherry-picked on top of topic/sof-dev to be tested.

### CODE REVIEW

**Comment by cujomalainey** [comment]
I appreciate these changes a lot. I am looking forward to having the hacks for the platform_name to platform conversion being removed.

**Comment by plbossart** [comment]
@cujomalainey can you check internally with your fellow Googlers if there was a reason to use a soc-soc-dummy platform name in broadwell and friends? Mark Brown tells me this is not needed, wondering why this was added a long time ago? That's really the only part that I'd like to sort out before pushing this upstream.

**Comment by cujomalainey** [comment]
I will follow up with the guys handling broadwell and see what they say. Let me blame the code and see who added it.

File: sound/soc/intel/boards/broadwell.c [hunk 1]
```diff
@@ -150,17 +150,44 @@ static int broadwell_rtd_init(struct snd_soc_pcm_runtime *rtd)
...
+	{
+		.name = "haswell-pcm-audio"
+	}
+};
+
+static struct snd_soc_dai_link_component platform_dummy_component[] = {
+	{
+		.name = "snd-soc-dummy"
+	}
+};
```
Comment (lyakh) [inline sound/soc/intel/boards/broadwell.c:177]: We're going to use identical dummy_codec_component[] and platform_dummy_component[] in multiple files, maybe define them once in a common file and use them from all? Many (distribution) kernel configurations build support for all platforms, but only one is used at run-time, so, even if those structs change at run-time, that shouldn't be a problem.

Comment (plbossart) [reply sound/soc/intel/boards/broadwell.c:177]: Yes, I've thought about this but at the same time if we add stuff to a common file then we'll get warnings that a structure is defined but not used. Using a global across multiple modules is also not great. I don't have a good solution so far.

**Comment by cujomalainey** [comment]
> I will follow up with the guys handling broadwell and see what they say. Let me blame the code and see who added it.

From fellow team members: 

> I remember these dummy names in DAI links were added for DPCM. Probably copied from https://www.kernel.org/doc/html/latest/sound/soc/dpcm.html
> "The FE DAI above sets the codec and code DAIs to dummy devices since the BE is dynamic and will change depending on runtime config."

Otherwise I don't have any concrete answers.

**Comment by plbossart** [comment]
@cujomalainey thanks. The explanations make sense for codec and dai_name, but for platform_name it's quite unclear. I'll give it a try on Broadwell (Samus) without setting the dummy platform name and remove it if I don't see a difference

File: sound/soc/intel/boards/broadwell.c (hunk 1 above)
Comment (bardliao) [reply sound/soc/intel/boards/broadwell.c:177]: Could we add it to a new created include/sound/soc-utils.h and include the soc-utils.h in the machine driver which need this?

Comment (plbossart) [reply sound/soc/intel/boards/broadwell.c:177]: I'll just try to remove the platform_name first, and we'll see if we can remove the dummy_component for the codecs later.

**Comment by plbossart** [comment]
closing, will submit upstream first

**Pull request closed without merging**